docs/pages/guides/Formats.example.jsx

```diff
@@ -1,31 +1,29 @@
-import format from 'date-fns/format';
 import React, { useState } from 'react';
 import frLocale from 'date-fns/locale/fr';
 import DateFnsAdapter from '@material-ui/pickers/adapter/date-fns';
 import { DatePicker } from '@material-ui/pickers';
-import { MuiPickersUtilsProvider } from '@material-ui/pickers';
+import { LocalizationProvider } from '@material-ui/pickers';
 
-class LocalizedUtils extends DateFnsAdapter {
-  getDatePickerHeaderText(date) {
-    return format(date, 'd MMM yyyy', { locale: this.locale });
-  }
-}
+/** @type Partial<import('@date-io/core/IUtils').DateIOFormats> */
+const formats = {
+  normalDate: 'd MMM yyy',
+  keyboardDate: 'd MMM yyy',
+};
 
 function DateFnsLocalizationExample() {
   const [selectedDate, handleDateChange] = useState(new Date());
 
   return (
-    <MuiPickersUtilsProvider utils={LocalizedUtils} locale={frLocale}>
+    <LocalizationProvider adapter={DateFnsAdapter} locale={frLocale} libFormats={formats}>
```

Comment:
Maybe to make it clear it's about formating dates?

```suggestion
    <LocalizationProvider adapter={DateFnsAdapter} locale={frLocale} dateFormats={formats}>
```

(If we were to anticipate a future use case where we would handle numbers and localization strings?)